README.md

```diff
@@ -183,214 +169,61 @@ require('telescope').setup{
 EOF
 ```
 
-<!-- TODO: move some options to Options affecting behavior -->
```

Comment:
Do you think moving defaults (except the minimal config setup) to [CONFIG.md](https://github.com/neovim/nvim-lspconfig/blob/master/CONFIG.md) like nvim-lspconfig would clean up things or clutter them?
The creation of the vimdoc page/help page could then be automatized with the approach taken from the [nvim-lua-guide](https://github.com/nanotee/nvim-lua-guide/blob/master/doc/nvim-lua-guide.txt).

Reply:
Plan was not to have any default config values in README. So currently we have the whole config in the readme and people just assume that they need to copy it in their config but they dont. They only need to change the values they wanna change. Also these options, mentioned in the readme, were not maintained for a while.

We have a way better system for that docgen. In the readme we teach how to change values and how `telescope.setup` works and then see `:help telescope.defaults` to the opts telescope provides and the defaults. Because we generate vimdoc, if someone changes something [here for example](https://github.com/nvim-telescope/telescope.nvim/pull/1024/files?file-filters%5B%5D=.lua&file-filters%5B%5D=.txt&file-filters%5B%5D=No+extension#diff-4936325bfc521d7cffc09fe0156becd4a4ba2ed169431c94bd63669fe0cc1a2aR130-R139) people usually also change the docs 5 lines below.

> The creation of the vimdoc page/help page could then be automatized with the approach taken from the nvim-lua-guide.

Already automated with a system that tj and i are maintaining https://github.com/tjdevries/tree-sitter-lua